wolfcrypt/src/chacha20_poly1305.c

```diff
@@ -167,6 +161,17 @@ int wc_ChaCha20Poly1305_Decrypt(
     return err;
 }
 
+int wc_ChaCha20Poly1305_CheckTag(
+    const byte authTag[CHACHA20_POLY1305_AEAD_AUTHTAG_SIZE],
+    const byte authTagChk[CHACHA20_POLY1305_AEAD_AUTHTAG_SIZE])
+{
+    int ret = 0;
```

Comment:
check values of routine input parameters for NULL.  
It's debatable but I'd consider changing the input params to pointers. The same comment applies everywhere where we have array arguments.  I agree with the opinion of Linus Torvalds as a bad C practice. 
https://stackoverflow.com/questions/54185824/is-using-array-arguments-in-c-considered-bad-practice

Reply:
Added check

Reply:
The use of the argument with fixed sizes `const byte authTagChk[CHACHA20_POLY1305_AEAD_AUTHTAG_SIZE]` is to keep compatibility with the existing code. I could remove it for the new API's, but I would also need to add a size argument and checks for that. Using it in the argument allows compiler time checks for the buffer sizes. 
@toddouska : Do you have a preference?